### ISSUE TITLE
_.over* enhancements

### ISSUE THREAD

**Issue by mvastola** [lodash/lodash#3987]
Two simple suggestions for the `_.over*` Utility functions (`_.over`, `_.overEvery` and `_.overSome`):
1. All of these currently take an array as there one and only argument. Could these functions also accept multiple arguments in lieu of an array? It would simplify the syntax a bit (for instance, in one case I have: `_.remove(args, _.overSome([_.isRegExp, _.isString]))` -- it'd be nice to ditch the square brackets.

2. In addition and/or instead, would it be possible for the `_.over*` functions to accept an optional value to process and -- if one is provided -- return the result of the that value passed to the function that is currently returned?
 
    In other words, if it would be nice to be able to call (assuming the first suggestion is also implemented) `_.overSome(arg, _.isRegExp, _.isString)` and get a truthy value returned. This could be done by checking if the first (or last, depending on your preference) argument is not a function (and either the other arguments are functions -- assuming my first suggestion -- or else there is exactly one other argument and it is an array). In the case where the value to be checked is a function as well, the intent can be clarified by passing the functions in an array as the second argument.

**Comment by jdalton** [comment]
Hi @mvastola!

I want to avoid more overloading so would prefer lower level methods that `_over*` could use. Can you think of what they would be?

**Comment by mvastola** [comment]
Could you clarify what you mean by that? Also, do you mean this for the first suggestion, the second, or both?

**Comment by jdalton** [comment]
@mvastola 
> Could you clarify what you mean by that?

To your first I may switch to accepting individual arguments since it can work with spread for the array case `_.overSome(...array)` but I'll noodle on that for a bit. No promises.

In response to your second suggestion about overloading the `_.over*` methods. I would want to not overload but instead try to create the primitive method that could create `_.over*` or the variations.

**Comment by mvastola** [comment]
I was mostly confused what you meant by 'primitive'. Do you mean you're looking to create an internal method that will do some of the heavy lifting and then create new public functions based on the functionality desired?

For the first suggestion, can't you just check if the first argument is an array and -- if so, use that -- and if not, use `[...arguments]`?

**Comment by jdalton** [comment]
@mvastola 

> Do you mean you're looking to create an internal method that will do some of the heavy lifting and then create new public functions based on the functionality desired?

I mean the second. Expose a lower level method that can create `_.over*`.

**Comment by jdalton** [comment]
**Update:**

Closing for now. Please feel free to pop-in and push this along a bit more.
